### ISSUE TITLE
Resque Redis Keys Not Expiring

### ISSUE THREAD

**Issue by jamesstonehill** [resque/resque#1689]
We're seeing the memory in our Redis instance increase gradually over time and the cause seems to be the these Resque keys that are not ever expiring. `resque:SEMAPHORE:lock:model_name_12345:AVAILABLE`

When I run in redis-cli `ttl resque:SEMAPHORE:lock:model_name_12345:AVAILABLE` I get `-1` to indicate there is no TTL for this key.

Is there a reason why these keys are set to never expire?

Gem version: 1.27.4

**Comment by codykaup** [comment]
We had the same issue. Turns out, it’s a default of Redis:

https://redis.io/commands/expire

> Normally Redis keys are created without an associated time to live. The key will simply live forever, unless it is removed by the user in an explicit way, for instance using the DEL command.

We did some temporary cleanup to mitigate the issue and are working through applying timeouts for our jobs.

**Comment by jamesstonehill** [comment]
Yeah, but why aren't these keys set with an expire? I'm not familiar with Resque internals so I'm not sure if these keys should be getting deleted through some other process, but if not then it seems that this would cause an inevitable rise in memory usage over time.

**Comment by codykaup** [comment]
There’s likely a lot of assumptions that would have to be made for default expirations to be set in Resque. It makes more sense for the user to set specific timeouts per key instead of keys disappearing.

For example: what would be a good global timeout? A few hours? A month? A year? It entirely depends on the application and even the specific key that’s being used.

**Comment by codykaup** [comment]
Completely agree there!

I'm not sure where that behavior is controlled though. I'm not sure that feature exists at the moment. 😦
